salt/loader.py

```diff
@@ -1224,19 +1228,31 @@ def refresh_file_mapping(self):
         self.file_mapping = salt.utils.odict.OrderedDict()
 
         for mod_dir in self.module_dirs:
-            files = []
             try:
-                # Make sure we have a sorted listdir in order to have expectable override results
+                # Make sure we have a sorted listdir in order to have
+                # expectable override results
                 files = sorted(os.listdir(mod_dir))
             except OSError:
                 continue  # Next mod_dir
+            if six.PY3:
+                try:
+                    pycache_files = [
+                        os.path.join('__pycache__', x) for x in
+                        sorted(os.listdir(os.path.join(mod_dir, '__pycache__')))
+                    ]
+                except OSError:
+                    pass
+                else:
+                    files = pycache_files + files
```

Comment:
This may be more efficient if I did:

```python
pycache_files.extend(files)
files = pycache_files
```
That would save us from allocating a brand new list before assigning the result to `files`.